examples/jsm/loaders/GLTFLoader.js

```diff
@@ -3659,6 +3751,235 @@ class GLTFParser {
+			if ( animationPointerDebug )
+				console.log( 'Original path: ' + path );
+
+			if ( ! path ) {
+
+				console.warn( 'Invalid path', ext, target );
+				return;
+
+			}
+
+			if ( path.startsWith( '/materials/' ) )
+				type = TARGET_TYPE.material;
+			else if ( path.startsWith( '/extensions/KHR_lights_punctual/lights/' ) )
+				type = TARGET_TYPE.light;
+			else if ( path.startsWith( '/cameras/' ) )
+				type = TARGET_TYPE.camera;
+
+			targetId = this.tryResolveNodeId( path, type );
+			if ( targetId === null || isNaN( targetId ) ) {
+
+				console.warn( 'Failed resolving animation node id: ' + targetId, path );
+				return;
+
+			} else {
+
+				if ( animationPointerDebug ) console.log( 'Resolved node ID for ' + type, targetId );
+
+			}
+
+			// TODO could be parsed better
+			switch ( type ) {
+
+				case TARGET_TYPE.material:
+					const pathIndex = ( '/materials/' + targetId.toString() + '/' ).length;
+					const pathStart = path.substring( 0, pathIndex );
+					targetProperty = path.substring( pathIndex );
+
+					// TODO implement all properties and/or find a better way to have a good mapping here
+					switch ( targetProperty ) {
+
+						case 'baseColorFactor':
+							targetProperty = 'color';
+							break;
+						case 'roughnessFactor':
+							targetProperty = 'roughness';
+							break;
+						case 'metallicFactor':
+							targetProperty = 'metalness';
+							break;
+						case 'emissiveFactor':
+							targetProperty = 'emissive';
+							break;
+						case 'alphaCutoff':
+							targetProperty = 'alphaTest';
+							break;
+						case 'extensions/KHR_materials_emissive_strength/emissiveStrength':
+							targetProperty = 'emissiveIntensity';
+							break;
+						case 'baseColorTexture/extensions/KHR_texture_transform/scale':
+							targetProperty = 'map/repeat';
+							break;
+						case 'baseColorTexture/extensions/KHR_texture_transform/offset':
+							targetProperty = 'map/offset';
+							break;
+						case 'emissiveTexture/extensions/KHR_texture_transform/scale':
+							targetProperty = 'emissiveMap/repeat';
+							break;
+						case 'emissiveTexture/extensions/KHR_texture_transform/offset':
+							targetProperty = 'emissiveMap/offset';
+							break;
+
+					}
+
+					path = pathStart + targetProperty;
+					if ( animationPointerDebug ) console.log( 'PROPERTY PATH', pathStart, targetProperty, path );
+					break;
+
+				case TARGET_TYPE.node:
+					const pathIndexNode = ( '/nodes/' + targetId.toString() + '/' ).length;
+					const pathStartNode = path.substring( 0, pathIndexNode );
+					targetProperty = path.substring( pathIndexNode );
+
+					switch ( targetProperty ) {
+
+						case 'translation':
+							targetProperty = 'position';
+							break;
+						case 'rotation':
+							targetProperty = 'quaternion';
+							break;
+						case 'scale':
+							targetProperty = 'scale';
+							break;
+						case 'weights':
+							targetProperty = 'morphTargetInfluences';
+							break;
+
+						// TODO matrix
```

Comment:
I think animation is required to target TRS properties, not a matrix?

Reply:
Good question. @UX3D-nopper is `/matrix` intentionally included in KHR_animation_pointer?